src/component/contents/exploration/DraftEditorBlockNode.react.js

```diff
@@ -299,43 +249,39 @@ class DraftEditorBlockNode extends React.Component<Props> {
           child,
           blockRenderMap,
         );
+
         const elementProps = getElementPropsConfig(
           child,
           editorKey,
           offsetKey,
           blockStyleFn,
           customConfig,
         );
+
         const childProps = {
           ...this.props,
           tree: editorState.getBlockTree(key),
           blockProps: customConfig.customProps,
           offsetKey,
+          wrapperTemplate,
           block: child,
         };
 
-        acc.push(
-          React.createElement(
+        acc.push({
+          wrapperTemplate,
+          block: child,
+          element: React.createElement(
             Element,
             elementProps,
             <Component {...childProps} />,
           ),
-        );
-
-        if (
-          !wrapperTemplate ||
-          shouldNotAddWrapperElement(child, contentState)
-        ) {
-          return acc;
-        }
-
+        });
         // if we are here it means we are the last block
         // that has a wrapperTemplate so we should wrap itself
         // and all other previous siblings that share the same wrapper
-        applyWrapperElementToSiblings(wrapperTemplate, Element, acc);
-
         return acc;
       }, []);
+      children = wrapBlockNodes(children, contentState);
```

Comment:
in order for `shouldNotAddWrapperElement` to work we need contentState I'm afraid...

Reply:
Passing the contentState its ok, now with the new data design thats how we can get the linked blocks